### DISCUSSION TITLE
How to solve a rendering error

### DISCUSSION THREAD

**Discussion by beyond2060** [quarto-dev/quarto-cli#11499] [Q&A]
### Description
  
All of the pages on my website render apart from the main index.qmd file, which has a couple of listings on it. It gives the error as shown below. The problem is that I haven't made any changes to the file recently, so it must be one of the other pages causing it, possibly due to the listings. But I don't know which page, and it is a large site, which makes it very difficult to solve this problem. How do I go about solving this?

---
```bash
**quarto render index.qmd**

ERROR: TypeError: Cannot read properties of undefined (reading 'split')

Stack trace:
    at eval (eval at <anonymous> (file:///Applications/quarto/bin/quarto.js:3925:16), <anonymous>:23:22)
    at renderEjs (file:///Applications/quarto/bin/quarto.js:90595:18)
    at templateMarkdownHandler (file:///Applications/quarto/bin/quarto.js:97842:28)
    at markdownHandler (file:///Applications/quarto/bin/quarto.js:98795:24)
    at file:///Applications/quarto/bin/quarto.js:98697:31
    at Array.forEach (<anonymous>)
    at listingHtmlDependencies (file:///Applications/quarto/bin/quarto.js:98696:24)
    at async Object.formatExtras (file:///Applications/quarto/bin/quarto.js:99025:45)
    at async runPandoc (file:///Applications/quarto/bin/quarto.js:76573:63)
    at async renderPandoc (file:///Applications/quarto/bin/quarto.js:78003:26)
```

**Comment by mcanouil** [comment]
Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.
See <https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us>.

The method is the same for everything, you remove pieces one by one to narrow down where the issue comes from.
Based on the error you likely have a malformed YAML entry somewhere.

Because nothing was shared (repository?), it's unfortunately impossible to help more than that.

**Comment by cscheid** [comment]
Thanks for the report - Quarto should be providing more information than it is, so I consider this a Quarto bug. But I would like to be able to reproduce it before fixing.

What Quarto version are you running? I can't seem to identify it from the line numbers in quarto.js.

Also, do you have a link to a git repo that you can share?

**Comment by beyond2060** [comment]
Thanks for your quick responses. Sorry I will share more next time (I have my GitHub repository set to private because I do a lot of experimenting with my Quarto instance and I'm afraid of accidentally sharing keys, but I'll sort that out). 

I finally worked this out. In one of my subdirectories I had accidentally created an empty filename.qmd file. I think that was causing the listing on the home page to throw an error. 
Quarto version 1.6.33

**Reply by mcanouil** [reply]
> I finally worked this out. In one of my subdirectories I had accidentally created an empty filename.qmd file. I think that was causing the listing on the home page to throw an error.
> Quarto version 1.6.33

If you manage to make a simple reproducible example which validate your assumption that would be helpful.

I only quickly tried:

```sh
quarto create project blog issue-11499 Issue-11499
mkdir -p posts/empty
touch posts/empty/index.qmd
quarto render
```

This only produced a warning (no crash/error) that the listing is missing metadata. (using latest 1.6 RC)

**Comment by beyond2060** [comment]
Ok I've investigated this some more and have a minimal site set up that causes the error. I am using listing templates so have my own .ejs files, and I expect that is the cause of the error and why you are not seeing it. Do you still want to look at it? I don't want to waste your time.

**Reply by cscheid** [reply]
Yes, I'd love to see it. I'd like Quarto to provide a more meaningful error message in this case so that it becomes easier for other folks to solve similar problems they might find.

**Reply by beyond2060** [reply]
Here you go:
https://github.com/beyond2060/quarto-bug-reproduction
If you do 
`Quarto render index.qmd`
you should get the error. (I just updated to quarto-1.6.37 and I still get the error).

**Reply by cderv** [reply]
I think this is because you do use `item.title.split()`
https://github.com/beyond2060/quarto-bug-reproduction/blob/09e70356eac59bc03c474cd35467abe617ce7252/js/all-table.ejs#L14C6-L14C135

but your document does not have a `title`
https://github.com/beyond2060/quarto-bug-reproduction/blob/main/experimenting/2024-11-22T175152Z/index.qmd

so `item.title` is undefined; Hence the error from EJS rendering
> Cannot read properties of undefined (reading 'split')

Add a title to your listing document, and the error will disappear

**Reply by cscheid** [reply]
Tracking the error-improvement enhancement here: https://github.com/quarto-dev/quarto-cli/issues/11529

**Reply by beyond2060** [reply]
Thanks guys, amazing support. ⭐️
